docs/kratos/emails-sms/05_custom-email-templates.mdx

```diff
@@ -0,0 +1,347 @@
+Follow these steps:
+
+1. List all your projects and get the Identity Service config file:
+
+   ```shell
+   ## List all available projects
+   ory list projects
+
+   ## Get config
+   ory get identity-config <project-id> --format yaml > identity-config.yaml
+   ```
+2. Add your Base64-encoded custom message templates to the config:
+
+   ```yaml title="identity-config.yaml"
+   courier:
+     smtp:
+       from_name: MyProject via Ory
+     templates:
+       recovery:
+         invalid:
+           email:
+             body: {
+               "html": "base64://ENCODED_HTML_TEMPLATE",
+               "plaintext": "base64://ENCODED_PLAINTEXT_TEMPLATE"
+             }
+         valid:
+           email:
+             body: {}
+       verification:
+         invalid:
+           email:
+             body: {}
+         valid:
+           email:
+             body: {}
+    ```
+3. Upload the edited config:
+
+   ```shell
+   ory update identity-config <project-id> --file updated_config.yaml
+   ```
+
+:::warning
+
+In Ory Cloud, you can use only Base64-encoded templates. The system doesn't accept `http://` or `file://` URIs. If you provide
+templates using unsupported methods, the system falls back to using the default templates.
+
+:::
+
+  </TabItem>
+  <TabItem value="selfhosted" label="Self-Hosted Ory Kratos Config" default>
+
+In self-hosted Ory Kratos instances, you can add custom templates from `http://`, `file://`, and `base64://` URIs.
+The `plaintext` and `html` fields are mandatory when defining the `body` key. All other keys are optional and will always fallback
+to the built-in templates or to the `template_override_path` (if specified).
+
+```yaml title=path/to/kratos/config.yml
+smtp:
+  template_override_path: /conf/courier-template
+  # ...
+  templates:
+    verification:
+      valid:
+        email:
+          body:
+            html: https://some-remote-resource/gotmpl
+            plaintext: base64://SGV5IHlvdSBkZWNvZGVkIG1lIDop
+          # optional
+          subject: file://some-file/subject.gotmpl
+```
+
+:::info
+
+When self-hosting Ory Kratos, the system expects the templates at these paths:
+
+- `<kratos-root>/<template-root>/recovery/valid`
+- `<kratos-root>/<template-root>/recovery/invalid`
+- `<kratos-root>/<template-root>/verification/valid`
+- `<kratos-root>/<template-root>/verification/invalid`
+
+Read [this document](../self-hosted/mail-courier-selfhosted) to learn more about the email courier and the specifics of using
+templates in self-hosted Ory Kratos instances.
+
+:::
+
+  </TabItem>
+</Tabs>
+````
+
+## Creating templates
+
+Templates use the golang template engine in the `text/template` package for rendering the `email.subject.gotmpl` and
+`email.body.plaintext.gotmpl` templates, and the `html/template` package for rendering the `email.body.gotmpl` template.
+
+Learn more:
+
+- https://pkg.go.dev/text/template
+- https://pkg.go.dev/html/template
+
+<a name="spring-disabled"></a>
```

Comment:
does this do anything?

Reply:
Yes! It's an anchor that takes you from one Spring "tip" to another. It's placed in this line to ensure that the right content gets put in the center of the screen when someone uses it.